Stratis.Bitcoin/Base/ChainState.cs

```diff
@@ -17,7 +18,11 @@ public class ChainState
         private bool ibdLastResult;
 
         /// <summary>A collection of blocks that have been found to be invalid.</summary>
-        internal ConcurrentDictionary<uint256, uint256> InvalidBlocks;
+        /// <remarks>All access to this object has to be protected by <see cref="invalidBlocksLock"/>.</remarks>
+        private readonly HashSet<uint256> invalidBlocks;
```

Comment:
No no no!!! 👎

Reply:
:( concurrent dictionary is no no no with that space wasting list of values you never use

Reply:
replaced with CHS